
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://nf-cmgg.github.io/smallvariants/1.10.1/parameters/">
      
      
        <link rel="prev" href="../output/">
      
      
        <link rel="next" href="../CITATIONS/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.14">
    
    
      
        <title>Parameters - nf-cmgg/smallvariants</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.342714a4.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Inter:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Inter";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="slate" data-md-color-primary="custom" data-md-color-accent="custom">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#nf-cmggsmallvariants-pipeline-parameters" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
      </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="nf-cmgg/smallvariants" class="md-header__button md-logo" aria-label="nf-cmgg/smallvariants" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16"><path d="M2.5 1.75v11.5c0 .138.112.25.25.25h3.17a.75.75 0 0 1 0 1.5H2.75A1.75 1.75 0 0 1 1 13.25V1.75C1 .784 1.784 0 2.75 0h8.5C12.216 0 13 .784 13 1.75v7.736a.75.75 0 0 1-1.5 0V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25m13.274 9.537zl-4.557 4.45a.75.75 0 0 1-1.055-.008l-1.943-1.95a.75.75 0 0 1 1.062-1.058l1.419 1.425 4.026-3.932a.75.75 0 1 1 1.048 1.074M4.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5M4 7.75A.75.75 0 0 1 4.75 7h2a.75.75 0 0 1 0 1.5h-2A.75.75 0 0 1 4 7.75"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            nf-cmgg/smallvariants
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Parameters
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="custom" data-md-color-accent="custom"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3m0-7 2.39 3.42C13.65 5.15 12.84 5 12 5s-1.65.15-2.39.42zM3.34 7l4.16-.35A7.2 7.2 0 0 0 5.94 8.5c-.44.74-.69 1.5-.83 2.29zm.02 10 1.76-3.77a7.131 7.131 0 0 0 2.38 4.14zM20.65 7l-1.77 3.79a7.02 7.02 0 0 0-2.38-4.15zm-.01 10-4.14.36c.59-.51 1.12-1.14 1.54-1.86.42-.73.69-1.5.83-2.29zM12 22l-2.41-3.44c.74.27 1.55.44 2.41.44.82 0 1.63-.17 2.37-.44z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="custom" data-md-color-accent="custom"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m17.75 4.09-2.53 1.94.91 3.06-2.63-1.81-2.63 1.81.91-3.06-2.53-1.94L12.44 4l1.06-3 1.06 3zm3.5 6.91-1.64 1.25.59 1.98-1.7-1.17-1.7 1.17.59-1.98L15.75 11l2.06-.05L18.5 9l.69 1.95zm-2.28 4.95c.83-.08 1.72 1.1 1.19 1.85-.32.45-.66.87-1.08 1.27C15.17 23 8.84 23 4.94 19.07c-3.91-3.9-3.91-10.24 0-14.14.4-.4.82-.76 1.27-1.08.75-.53 1.93.36 1.85 1.19-.27 2.86.69 5.83 2.89 8.02a9.96 9.96 0 0 0 8.02 2.89m-1.64 2.02a12.08 12.08 0 0 1-7.8-3.47c-2.17-2.19-3.33-5-3.49-7.82-2.81 3.14-2.7 7.96.31 10.98 3.02 3.01 7.84 3.12 10.98.31"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91s2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/nf-cmgg/smallvariants" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8M97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    nf-cmgg/smallvariants
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="nf-cmgg/smallvariants" class="md-nav__button md-logo" aria-label="nf-cmgg/smallvariants" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16"><path d="M2.5 1.75v11.5c0 .138.112.25.25.25h3.17a.75.75 0 0 1 0 1.5H2.75A1.75 1.75 0 0 1 1 13.25V1.75C1 .784 1.784 0 2.75 0h8.5C12.216 0 13 .784 13 1.75v7.736a.75.75 0 0 1-1.5 0V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25m13.274 9.537zl-4.557 4.45a.75.75 0 0 1-1.055-.008l-1.943-1.95a.75.75 0 0 1 1.062-1.058l1.419 1.425 4.026-3.932a.75.75 0 1 1 1.048 1.074M4.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5M4 7.75A.75.75 0 0 1 4.75 7h2a.75.75 0 0 1 0 1.5h-2A.75.75 0 0 1 4 7.75"/></svg>

    </a>
    nf-cmgg/smallvariants
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/nf-cmgg/smallvariants" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8M97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    nf-cmgg/smallvariants
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../usage/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usage
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../output/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Output
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Parameters
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Parameters
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#inputoutput-options" class="md-nav__link">
    <span class="md-ellipsis">
      Input/output options
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reference-genome-options" class="md-nav__link">
    <span class="md-ellipsis">
      Reference genome options
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pipeline-specific-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Pipeline specific parameters
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#institutional-config-options" class="md-nav__link">
    <span class="md-ellipsis">
      Institutional config options
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#generic-options" class="md-nav__link">
    <span class="md-ellipsis">
      Generic options
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#annotation-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Annotation parameters
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../CITATIONS/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Citations
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#inputoutput-options" class="md-nav__link">
    <span class="md-ellipsis">
      Input/output options
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reference-genome-options" class="md-nav__link">
    <span class="md-ellipsis">
      Reference genome options
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pipeline-specific-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Pipeline specific parameters
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#institutional-config-options" class="md-nav__link">
    <span class="md-ellipsis">
      Institutional config options
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#generic-options" class="md-nav__link">
    <span class="md-ellipsis">
      Generic options
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#annotation-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Annotation parameters
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="nf-cmggsmallvariants-pipeline-parameters">nf-cmgg/smallvariants pipeline parameters</h1>
<p>A nextflow pipeline for calling and annotating small variants from short DNA reads for WES and WGS data</p>
<h2 id="inputoutput-options">Input/output options</h2>
<p>Define where the pipeline should find input data and save output data.</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Type</th>
<th>Default</th>
<th>Required</th>
<th>Hidden</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>input</code></td>
<td>Path to comma-separated file containing information about the samples in the experiment. <details><summary>Help</summary><small>You will need to create a design file with information about the samples in your experiment before running the pipeline. Use this parameter to specify its location. It has to be a comma-separated file with samples, and a header row. See <a href="../usage/">usage docs</a>.</small></details></td>
<td><code>string</code></td>
<td></td>
<td>True</td>
<td></td>
</tr>
<tr>
<td><code>outdir</code></td>
<td>The output directory where the results will be saved. You have to use absolute paths to storage on Cloud infrastructure.</td>
<td><code>string</code></td>
<td></td>
<td>True</td>
<td></td>
</tr>
<tr>
<td><code>watchdir</code></td>
<td>A folder to watch for the creation of files that start with <code>watch:</code> in the samplesheet.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>email</code></td>
<td>Email address for completion summary. <details><summary>Help</summary><small>Set this parameter to your e-mail address to get a summary e-mail with details of the run sent to you when the workflow exits. If set in your user config file (<code>~/.nextflow/config</code>) then you don't need to specify this on the command line for every run.</small></details></td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>ped</code></td>
<td>Path to a pedigree file for all samples in the run. All relational data will be fetched from this file.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<h2 id="reference-genome-options">Reference genome options</h2>
<p>Reference genome related files and options required for the workflow.</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Type</th>
<th>Default</th>
<th>Required</th>
<th>Hidden</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>genome</code></td>
<td>Reference genome build. Used to fetch the right reference files. <details><summary>Help</summary><small>Requires a Genome Reference Consortium reference ID (e.g. GRCh38)</small></details></td>
<td><code>string</code></td>
<td>GRCh38</td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>fasta</code></td>
<td>Path to FASTA genome file. <details><summary>Help</summary><small>This parameter is <em>mandatory</em> if <code>--genome</code> is not specified. The path to the reference genome fasta.</small></details></td>
<td><code>string</code></td>
<td></td>
<td>True</td>
<td></td>
</tr>
<tr>
<td><code>fai</code></td>
<td>Path to FASTA genome index file.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>dict</code></td>
<td>Path to the sequence dictionary generated from the FASTA reference. This is only used when <code>haplotypecaller</code> is one of the specified callers.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>strtablefile</code></td>
<td>Path to the STR table file generated from the FASTA reference. This is only used when <code>--dragstr</code> has been given.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>sdf</code></td>
<td>Path to the SDF folder generated from the reference FASTA file. This is only required when using <code>--validate</code>.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>elfasta</code></td>
<td>Path to the ELFASTA genome file. This is used when <code>elprep</code> is part of the callers and will be automatically generated when missing.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>elsites</code></td>
<td>Path to the elsites file. This is used when <code>elprep</code> is part of the callers.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>genomes</code></td>
<td>Object for genomes</td>
<td><code>object</code></td>
<td></td>
<td></td>
<td>True</td>
</tr>
<tr>
<td><code>genomes_base</code></td>
<td>Directory base for CMGG reference store (used when <code>--genomes_ignore false</code> is specified)</td>
<td><code>string</code></td>
<td>/references/</td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>cmgg_config_base</code></td>
<td>The base directory for the local config files</td>
<td><code>string</code></td>
<td>/conf/</td>
<td></td>
<td>True</td>
</tr>
<tr>
<td><code>genomes_ignore</code></td>
<td>Do not load the local references from the path specified with <code>--genomes_base</code></td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td>True</td>
</tr>
<tr>
<td><code>igenomes_base</code></td>
<td>Directory / URL base for iGenomes references.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td>True</td>
</tr>
<tr>
<td><code>igenomes_ignore</code></td>
<td>Do not load the iGenomes reference config. <details><summary>Help</summary><small>Do not load <code>igenomes.config</code> when running the pipeline. You may choose this option if you observe clashes between custom parameters and those supplied in <code>igenomes.config</code>.</small></details></td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td>True</td>
</tr>
</tbody>
</table>
<h2 id="pipeline-specific-parameters">Pipeline specific parameters</h2>
<p>Parameters that define how the pipeline works</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Type</th>
<th>Default</th>
<th>Required</th>
<th>Hidden</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>scatter_count</code></td>
<td>The amount of scattering that should happen per sample. <details><summary>Help</summary><small>Increase this number to increase the pipeline run speed, but at the tradeoff of using more IO and disk space. This can differ from the actual scatter count in some cases (especially with smaller files).<br>This has an effect on HaplotypeCaller, GenomicsDBImport and GenotypeGVCFs.</small></details></td>
<td><code>integer</code></td>
<td>40</td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>merge_distance</code></td>
<td>The merge distance for family BED files <details><summary>Help</summary><small>Increase this parameter if GenomicsDBImport is running slow. This defines the maximum distance between intervals that should be merged. The less intervals GenomicsDBImport actually gets, the faster it will run.</small></details></td>
<td><code>integer</code></td>
<td>100000</td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>dragstr</code></td>
<td>Create DragSTR models to be used with HaplotypeCaller <details><summary>Help</summary><small>This currently is only able to run single-core per sample. Due to this, the process is very slow with only very small improvements to the analysis.</small></details></td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>validate</code></td>
<td>Validate the found variants</td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>filter</code></td>
<td>Filter the found variants.</td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>annotate</code></td>
<td>Annotate the found variants using Ensembl VEP.</td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>add_ped</code></td>
<td>Add PED INFO header lines to the final VCFs.</td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>gemini</code></td>
<td>Create a Gemini databases from the final VCFs.</td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>mosdepth_slow</code></td>
<td>Don't run mosdepth in fast-mode <details><summary>Help</summary><small>This is advised if you need exact coverage BED files as output.</small></details></td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>roi</code></td>
<td>Path to the default ROI (regions of interest) BED file to be used for WES analysis. <details><summary>Help</summary><small>This will be used for all samples that do not have a specific ROI file supplied to them through the samplesheet. Don't supply an ROI file to run the analysis as WGS.</small></details></td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>dbsnp</code></td>
<td>Path to the dbSNP VCF file. This will be used to set the variant IDs.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>dbsnp_tbi</code></td>
<td>Path to the index of the dbSNP VCF file.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>somalier_sites</code></td>
<td>Path to the VCF file with sites for Somalier to use.</td>
<td><code>string</code></td>
<td>https://github.com/brentp/somalier/files/3412456/sites.hg38.vcf.gz</td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>only_call</code></td>
<td>Only call the variants without doing any post-processing.</td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>only_merge</code></td>
<td>Only run the pipeline until the creation of the genomicsdbs and output them.</td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>output_genomicsdb</code></td>
<td>Output the genomicsDB together with the joint-genotyped VCF.</td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>callers</code></td>
<td>A comma delimited string of the available callers. Current options are: <code>haplotypecaller</code> and <code>vardict</code>.</td>
<td><code>string</code></td>
<td>haplotypecaller</td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>vardict_min_af</code></td>
<td>The minimum allele frequency for VarDict when no <code>vardict_min_af</code> is supplied in the samplesheet.</td>
<td><code>number</code></td>
<td>0.1</td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>normalize</code></td>
<td>Normalize the variant in the final VCFs.</td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>only_pass</code></td>
<td>Filter out all variants that don't have the PASS filter for vardict. This only works when <code>--filter</code> is also given.</td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>keep_alt_contigs</code></td>
<td>Keep all aditional contigs for calling instead of filtering them out before.</td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>updio</code></td>
<td>Run UPDio analysis on the final VCFs.</td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>updio_common_cnvs</code></td>
<td>A TSV file containing common CNVs to be used by UPDio.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>automap</code></td>
<td>Run AutoMap analysis on the final VCFs.</td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>automap_repeats</code></td>
<td>BED file with repeat regions in the genome. <details><summary>Help</summary><small>This file will be automatically generated for hg38/GRCh38 and hg19/GRCh37 when this parameter has not been given.</small></details></td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>automap_panel</code></td>
<td>TXT file with gene panel regions to be used by AutoMap. <details><summary>Help</summary><small>By default the CMGG gene panel list will be used.</small></details></td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>automap_panel_name</code></td>
<td>The panel name of the panel given with --automap_panel.</td>
<td><code>string</code></td>
<td>cmgg_bio</td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>hc_phasing</code></td>
<td>Perform phasing with HaplotypeCaller.</td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>min_callable_coverage</code></td>
<td>The lowest callable coverage to determine callable regions.</td>
<td><code>integer</code></td>
<td>5</td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>unique_out</code></td>
<td>Don't change this value</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td>True</td>
</tr>
<tr>
<td><code>disable_hc_dict_validation</code></td>
<td>Disable the sequence dictionary validation in HaplotypeCaller</td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>skip_merged_cram_output</code></td>
<td>Don't output the merged CRAM files.</td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<h2 id="institutional-config-options">Institutional config options</h2>
<p>Parameters used to describe centralised config profiles. These should not be edited.</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Type</th>
<th>Default</th>
<th>Required</th>
<th>Hidden</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>custom_config_version</code></td>
<td>Git commit id for Institutional configs.</td>
<td><code>string</code></td>
<td>master</td>
<td></td>
<td>True</td>
</tr>
<tr>
<td><code>custom_config_base</code></td>
<td>Base directory for Institutional configs. <details><summary>Help</summary><small>If you're running offline, Nextflow will not be able to fetch the institutional config files from the internet. If you don't need them, then this is not a problem. If you do need them, you should download the files from the repo and tell Nextflow where to find them with this parameter.</small></details></td>
<td><code>string</code></td>
<td>https://raw.githubusercontent.com/nf-core/configs/master</td>
<td></td>
<td>True</td>
</tr>
<tr>
<td><code>config_profile_name</code></td>
<td>Institutional config name.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td>True</td>
</tr>
<tr>
<td><code>config_profile_description</code></td>
<td>Institutional config description.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td>True</td>
</tr>
<tr>
<td><code>config_profile_contact</code></td>
<td>Institutional config contact information.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td>True</td>
</tr>
<tr>
<td><code>config_profile_url</code></td>
<td>Institutional config URL link.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td>True</td>
</tr>
</tbody>
</table>
<h2 id="generic-options">Generic options</h2>
<p>Less common options for the pipeline, typically set in a config file.</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Type</th>
<th>Default</th>
<th>Required</th>
<th>Hidden</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>version</code></td>
<td>Display version and exit.</td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>publish_dir_mode</code></td>
<td>Method used to save pipeline results to output directory. <details><summary>Help</summary><small>The Nextflow <code>publishDir</code> option specifies which intermediate files should be saved to the output directory. This option tells the pipeline what method should be used to move these files. See <a href="https://www.nextflow.io/docs/latest/process.html#publishdir">Nextflow docs</a> for details.</small></details></td>
<td><code>string</code></td>
<td>copy</td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>email_on_fail</code></td>
<td>Email address for completion summary, only when pipeline fails. <details><summary>Help</summary><small>An email address to send a summary email to when the pipeline is completed - ONLY sent if the pipeline does not exit successfully.</small></details></td>
<td><code>string</code></td>
<td></td>
<td></td>
<td>True</td>
</tr>
<tr>
<td><code>plaintext_email</code></td>
<td>Send plain-text email instead of HTML.</td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td>True</td>
</tr>
<tr>
<td><code>max_multiqc_email_size</code></td>
<td>File size limit when attaching MultiQC reports to summary emails.</td>
<td><code>string</code></td>
<td>25.MB</td>
<td></td>
<td>True</td>
</tr>
<tr>
<td><code>monochrome_logs</code></td>
<td>Do not use coloured log outputs.</td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td>True</td>
</tr>
<tr>
<td><code>hook_url</code></td>
<td>Incoming hook URL for messaging service <details><summary>Help</summary><small>Incoming hook URL for messaging service. Currently, MS Teams and Slack are supported.</small></details></td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>multiqc_title</code></td>
<td>MultiQC report title. Printed as page header, used for filename if not otherwise specified.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>multiqc_config</code></td>
<td>Custom config file to supply to MultiQC.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>multiqc_logo</code></td>
<td>Custom logo file to supply to MultiQC. File name must also be set in the MultiQC config file</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>multiqc_methods_description</code></td>
<td>Custom MultiQC yaml file containing HTML including a methods description.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>validate_params</code></td>
<td>Boolean whether to validate parameters against the schema at runtime</td>
<td><code>boolean</code></td>
<td>True</td>
<td></td>
<td>True</td>
</tr>
<tr>
<td><code>pipelines_testdata_base_path</code></td>
<td>Base URL or local path to location of pipeline test dataset files</td>
<td><code>string</code></td>
<td>https://raw.githubusercontent.com/nf-core/test-datasets/</td>
<td></td>
<td>True</td>
</tr>
</tbody>
</table>
<h2 id="annotation-parameters">Annotation parameters</h2>
<p>Parameters to configure Ensembl VEP and VCFanno</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Type</th>
<th>Default</th>
<th>Required</th>
<th>Hidden</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>vep_chunk_size</code></td>
<td>The amount of sites per split VCF as input to VEP.</td>
<td><code>integer</code></td>
<td>50000</td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>species</code></td>
<td>The species of the samples. <details><summary>Help</summary><small>Must be lower case and have underscores as spaces.</small></details></td>
<td><code>string</code></td>
<td>homo_sapiens</td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>vep_merged</code></td>
<td>Specify if the VEP cache is a merged cache.</td>
<td><code>boolean</code></td>
<td>True</td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>vep_cache</code></td>
<td>The path to the VEP cache.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>vep_dbnsfp</code></td>
<td>Use the dbNSFP plugin with Ensembl VEP. <details><summary>Help</summary><small>The '--dbnsfp' and '--dbnsfp_tbi' parameters need to be specified when using this parameter.</small></details></td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>vep_spliceai</code></td>
<td>Use the SpliceAI plugin with Ensembl VEP. <details><summary>Help</summary><small>The '--spliceai_indel', '--spliceai_indel_tbi', '--spliceai_snv' and '--spliceai_snv_tbi' parameters need to be specified when using this parameter.</small></details></td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>vep_spliceregion</code></td>
<td>Use the SpliceRegion plugin with Ensembl VEP.</td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>vep_mastermind</code></td>
<td>Use the Mastermind plugin with Ensembl VEP. <details><summary>Help</summary><small>The '--mastermind' and '--mastermind_tbi' parameters need to be specified when using this parameter.</small></details></td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>vep_maxentscan</code></td>
<td>Use the MaxEntScan plugin with Ensembl VEP. <details><summary>Help</summary><small>The '--maxentscan' parameter need to be specified when using this parameter.</small></details></td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>vep_eog</code></td>
<td>Use the custom EOG annotation with Ensembl VEP. <details><summary>Help</summary><small>The '--eog' and '--eog_tbi' parameters need to be specified when using this parameter.</small></details></td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>vep_alphamissense</code></td>
<td>Use the AlphaMissense plugin with Ensembl VEP. <details><summary>Help</summary><small>The '--alphamissense' and '--alphamissense_tbi' parameters need to be specified when using this parameter.</small></details></td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>vep_version</code></td>
<td>The version of the VEP tool to be used.</td>
<td><code>number</code></td>
<td>105.0</td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>vep_cache_version</code></td>
<td>The version of the VEP cache to be used.</td>
<td><code>integer</code></td>
<td>105</td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>dbnsfp</code></td>
<td>Path to the dbSNFP file.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>dbnsfp_tbi</code></td>
<td>Path to the index of the dbSNFP file.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>spliceai_indel</code></td>
<td>Path to the VCF containing indels for spliceAI.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>spliceai_indel_tbi</code></td>
<td>Path to the index of the VCF containing indels for spliceAI.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>spliceai_snv</code></td>
<td>Path to the VCF containing SNVs for spliceAI.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>spliceai_snv_tbi</code></td>
<td>Path to the index of the VCF containing SNVs for spliceAI.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>mastermind</code></td>
<td>Path to the VCF for Mastermind.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>mastermind_tbi</code></td>
<td>Path to the index of the VCF for Mastermind.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>alphamissense</code></td>
<td>Path to the TSV for AlphaMissense.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>alphamissense_tbi</code></td>
<td>Path to the index of the TSV for AlphaMissense.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>eog</code></td>
<td>Path to the VCF containing EOG annotations.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>eog_tbi</code></td>
<td>Path to the index of the VCF containing EOG annotations.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>vcfanno</code></td>
<td>Run annotations with vcfanno.</td>
<td><code>boolean</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>vcfanno_config</code></td>
<td>The path to the VCFanno config TOML.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>vcfanno_lua</code></td>
<td>The path to a Lua script to be used in VCFanno.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><code>vcfanno_resources</code></td>
<td>A semicolon-seperated list of resource files for VCFanno, please also supply their indices using this parameter.</td>
<td><code>string</code></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
    
      
      <nav class="md-footer__inner md-grid" aria-label="Footer" >
        
          
          <a href="../output/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Output">
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
            </div>
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Previous
              </span>
              <div class="md-ellipsis">
                Output
              </div>
            </div>
          </a>
        
        
          
          <a href="../CITATIONS/" class="md-footer__link md-footer__link--next" aria-label="Next: Citations">
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Next
              </span>
              <div class="md-ellipsis">
                Citations
              </div>
            </div>
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11z"/></svg>
            </div>
          </a>
        
      </nav>
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "..", "features": ["content.code.annotate", "content.code.copy", "navigation.instant", "navigation.top", "navigation.tracking", "navigation.sections", "navigation.footer", "search.share", "toc.follow"], "search": "../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"provider": "mike"}}</script>
    
    
      <script src="../assets/javascripts/bundle.13a4f30d.min.js"></script>
      
    
  </body>
</html>